UnityProject/Assets/Scripts/Core/Editor/QuickBuild.cs

```diff
@@ -0,0 +1,449 @@
+			if (pathForDisplay.IndexOf(projectPath, 0, projectPath.Length) == 0)
+			{
+				var basePath = Path.GetFullPath(Path.Combine(projectPath, "..", ".."));
+				if (basePath.EndsWith(Path.DirectorySeparatorChar.ToString()) == false)
+				{
+					basePath += Path.DirectorySeparatorChar;
+				}
+				pathForDisplay = pathForDisplay.Remove(0, basePath.Length);
+			}
+		}
+
+		private void SelectPath()
+		{
+			var userInput = EditorUtility.SaveFolderPanel("Build Destination", buildPath, "");
+			if (string.IsNullOrEmpty(userInput)) return; // Canceling the dialogue retuns null
+
+			// Unity returns linux separator; instead use platform-specific separator
+			buildPath = userInput.Replace('/', Path.DirectorySeparatorChar);
+			SetPathForDisplay(buildPath);
+		}
+
+		private bool BuildButton()
+		{
+			GUIStyle buildBtnStyle = new GUIStyle(GUI.skin.button);
+			buildBtnStyle.padding = new RectOffset(30, 30, 8, 8);
+
+			GUIContent btnTxt = new GUIContent("Build");
+			Rect rect = GUILayoutUtility.GetRect(btnTxt, buildBtnStyle, GUILayout.ExpandWidth(false));
+			rect.center = new Vector2(EditorGUIUtility.currentViewWidth / 2, rect.center.y);
+
+			return GUI.Button(rect, btnTxt, GUI.skin.button);
+		}
+
+		private void UpdateGameManager(bool isQuickLoad)
+		{
+			if (gameManager == null)
+			{
+				Logger.LogWarning($"{nameof(GameManager)} not found! Cannot set {nameof(GameManager.QuickLoad)} property.");
+				return;
+			}
+
+			if (gameManager.QuickLoad != isQuickLoad)
+			{
+				gameManager.QuickLoad = isQuickLoad;
+				EditorUtility.SetDirty(gameManager);
+				AssetDatabase.SaveAssets();
+			}
+		}
+
+		private void Build()
+		{
+			UpdateGameManager(isQuickLoad);
+
+			string[] sceneNames = new string[requiredScenes.Length + 1];
+			sceneNames[sceneNames.Length - 1] = mainStationScene;
+			var scenePaths = EditorBuildSettings.scenes
+					.Where(s => sceneNames.Contains(Path.GetFileNameWithoutExtension(s.path)))
+					.Select(s => s.path);
+
+			BuildPlayerOptions buildPlayerOptions = new BuildPlayerOptions
+			{
+				scenes = scenePaths.ToArray(),
+				locationPathName = buildPath,
+				target = target,
+				options = BuildOptions.None
+			};
+
+			if (isDevelopmentBuild)
+			{
+				buildPlayerOptions.options = BuildOptions.Development;
+				if (isScriptsOnly)
+				{
+					buildPlayerOptions.options |= BuildOptions.Development;
+				}
+			}
+
+			BuildReport report = BuildPipeline.BuildPlayer(buildPlayerOptions);
+			var timeStr = $"{report.summary.totalTime.Hours}:{report.summary.totalTime.Minutes}:{report.summary.totalTime.Seconds}";
+
+			if (report.summary.result == BuildResult.Succeeded)
+			{
+				Logger.Log($"Build complete. ({timeStr})");
+			}
+
+			if (report.summary.result == BuildResult.Failed)
+			{
+				Logger.LogError($"Build failed! ({timeStr})");
+			}
+		}
+
+		#endregion
+
+		#region Disable Scenes Tab
+
+		private static List<string> AdditionalScenesToRemove = new List<string>()
+		{
+			"Fallstation Centcom",
```

Comment:
I remember NaughtyAttributes had  an attribute to work with scenes in inspector while internally keep using strings.

Reply:
aaaah but I imagine naughtyattributes doesn't work with this custom window, right?

Reply:
Yep, I'm using it. Ignore the Disable Scenes region - that's existing code that's just been namespaced.